src/tutorial.md

```diff
@@ -0,0 +1,1056 @@
+        (total-due (+ (get amount current-loan-details) (unwrap! accrued-interest (err u8))))
+        (new-loan (+ total-due amount))
+    )
+        (asserts! (<= amount allowed-borrow) (err u7))
+        (try! (contract-call? .asset transfer amount (as-contract tx-sender) tx-sender none))
+        (map-set loans { owner: tx-sender } { amount: new-loan, last-interaction-block: block-height })
+        (ok true)
+    )
+)
+```
+
+Again here we are using `let` to retrieve and set the amount this user currently has in the pool, defaulting to 0.
+
+Next we are getting the number that this user is allowed to borrow. Borrowers can only borrow up to 50% of their collateral.
+
+Next we are checking to see how many blocks it has been since they last interacted with the protocol. We set this to keep track of interest accrual. We use the `calculate-accrued-interest` function (which we'll implement in a moment) to calculate how much they owe and set that as well.
+
+Next we are checking to make sure that the depositor is allowed to borrow and that the pool has enough liquidity to cover the new loan.
+
+Then we transfer the sBTC from the contract to the borrower.
+
+Finally we update all of our variables to include the new amount. Note that when we update these variables, we are taking interest into account. In this case, when this borrower repays this loan, they will repay the original amount plus the 10% interest.
+
+Next up we have the repay function.
+
+```clojure
+;; Users can repay their sBTC loans
+(define-public (repay (amount uint))
+    (let (
+        (current-loan-details (default-to { amount: u0, last-interaction-block: u0 } (map-get? loans { owner: tx-sender })))
+        (accrued-interest (calculate-accrued-interest (get amount current-loan-details) (get last-interaction-block current-loan-details)))
+        (total-due (+ (get amount current-loan-details) (unwrap! accrued-interest (err u8))))
+    )
+        (asserts! (>= total-due amount) (err u4))
+        (try! (contract-call? .asset transfer amount tx-sender (as-contract tx-sender) none))
+        (map-set loans { owner: tx-sender } { amount: (- total-due amount), last-interaction-block: block-height })
+        (var-set total-loans (- (var-get total-loans) amount))
+        (ok true)
+    )
+)
+```
+
+This is very similar to the other functions. In this case the borrower is simply paying back their loan with the included interest. Remember that the interest was calculated when they initially borrowed, so we don't need to include it here.
+
+Now let's write the function that allows lenders to claim their yield.
+
+```clojure
+(define-public (claim-yield)
+    (let (
+        (user-deposit (default-to u0 (get amount (map-get? deposits { owner: tx-sender }))))
+        (yield-amount (/ (* (var-get pool-reserve) user-deposit) (var-get total-deposits)))
+    )
+        (try! (contract-call? .asset transfer yield-amount (as-contract tx-sender) tx-sender none))
```

Comment:
contract-call needs to be wrapped in `as-contract` and sbtc would be transferred to owner